src/containers/views/DataFiles.js

```diff
@@ -1,28 +1,43 @@
 import React, { Component, PropTypes } from 'react';
-import { Link } from 'react-router';
+import { browserHistory, withRouter, Link } from 'react-router';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
 import _ from 'underscore';
+import Breadcrumbs from '../../components/Breadcrumbs';
 import { getDeleteMessage, getNotFoundMessage } from '../../constants/lang';
 import InputSearch from '../../components/form/InputSearch';
 import Button from '../../components/Button';
 import { fetchDataFiles, deleteDataFile } from '../../actions/datafiles';
 import { search } from '../../actions/utils';
 import { filterByFilename } from '../../reducers/datafiles';
+import { getFilenameFromPath } from '../../utils/helpers';
 import { ADMIN_PREFIX } from '../../constants';
 
 export class DataFiles extends Component {
 
   componentDidMount() {
-    const { fetchDataFiles } = this.props;
-    fetchDataFiles();
+    const { fetchDataFiles, params } = this.props;
+    fetchDataFiles(params.splat);
   }
 
-  handleClickDelete(filename) {
-    const { deleteDataFile } = this.props;
-    const confirm = window.confirm(getDeleteMessage(filename));
+  componentWillReceiveProps(nextProps) {
+    const { fetchDataFiles } = nextProps;
+    if (this.props.params.splat !== nextProps.params.splat) {
+      fetchDataFiles(nextProps.params.splat);
+    }
+  }
+
+  handleClickDelete(path) {
+    const { deleteDataFile, params } = this.props;
+    const confirm = window.confirm(getDeleteMessage(path));
+    const directory = params.splat || "";
+    const dir = directory ? `/${directory}` : ""
```

Comment:
aargh.. I hate it that the linter doesn't report this while running `npm run test:watch`